### PR TITLE
Add default values when reading action request parameter

### LINKED ISSUE #14990: Controller/Component/AuthComponent.php declare(strict_types=1);
* [x] bug
* 4.x (master)

### What you did
// Instance reflection class of a controller (with AuthComponent)

$rcController = new \ReflectionClass($controllerClassName);
$obController = $rcController->newInstance();
$obController->startupProcess();

### What happened
Controller/Component/AuthComponent.php lines 271 and 330 TypeError.

Argument 1 passed to Cake\Controller\Controller::isAction() must be of the type string, null given, called in C:\wamp64\www\cakephp4\plugins\vendor\cakephp\cakephp\src\Controller\Component\AuthComponent.php on line 271

### Solution
Replace line 270
$action = $controller->getRequest()->getParam('action');
with 
$action = $controller->getRequest()->getParam('action', '');

Replace line 330
$action = strtolower($controller->getRequest()->getParam('action'));
with
$action = strtolower($controller->getRequest()->getParam('action', ''));

### PR DESCRIPTION
**Pull request by markstory**
Fixes #14990

### CODE REVIEW

**Review by othercorey — APPROVED** [review]
Looks ok

File: src/Controller/Component/AuthComponent.php [hunk 1]
```diff
@@ -267,7 +267,7 @@ public function authCheck(EventInterface $event): ?Response
         /** @var \Cake\Controller\Controller $controller */
         $controller = $event->getSubject();
 
-        $action = $controller->getRequest()->getParam('action');
+        $action = $controller->getRequest()->getParam('action', '');
         if (!$controller->isAction($action)) {
```
Comment (ndm2) [inline src/Controller/Component/AuthComponent.php:271]: How about `$action !== null &&` instead? Should be quite a bit faster than doing reflection on something that is already known to not exist?

File: src/Controller/Component/AuthComponent.php [hunk 2]
```diff
@@ -268,7 +268,7 @@ public function authCheck(EventInterface $event): ?Response
         $controller = $event->getSubject();
 
         $action = $controller->getRequest()->getParam('action');
-        if (!$controller->isAction($action)) {
+        if ($action !== null && !$controller->isAction($action)) {
```
Comment (dereuromark) [inline src/Controller/Component/AuthComponent.php:271]: Should this have been
`$action === null ||`
?

Comment (ndm2) [reply src/Controller/Component/AuthComponent.php:271]: It should have 😢

Comment (othercorey) [reply src/Controller/Component/AuthComponent.php:271]: It seems like we need a proper unit test for this.

**Pull request merged**